ansible/playbooks/roles/grafana/tasks/upgrade/main.yml

```diff
@@ -0,0 +1,81 @@
+---
+- name: Include new version of defaults
+  include_vars:
+    file: defaults/versions.yml
+    name: grafana_versions
+
+- name: Include upgrade defaults
+  include_vars:
+    file: defaults/upgrade.yml
+    name: upgrade_defaults
+
+# If state file exists it means the previous run failed
+- name: Check for upgrade state file
+  stat:
+    path: "{{ upgrade_defaults.state_file_path }}"
+    get_attributes: false
+    get_checksum: false
+    get_mime: false
+  register: state_file_status
+
+- name: Set grafana_will_upgrade fact
+  set_fact:
+    grafana_will_upgrade: false
+
+- name: Start Grafana upgrade
+  when:
+    - ansible_facts.services["grafana.service"] is defined
+    - grafana_stat.stat.exists
```

Comment:
The condition is incorrect since upgrade state file is created inside the nested block. I would simplify this to have such condition for main block:

```
  when: grafana_target_version is version(grafana_installed_version, '>')
     or state_file_status.stat.exists
```

Then `grafana_will_upgrade` would be unneeded.